src/scheduling.jl

```diff
@@ -1,18 +1,55 @@
 import Dagger
 using Distributed
+using Graphs
 using MetaGraphs
 
+mutable struct DataObject
+    data
+    size::Float64
+end
+
+function populate_data_object!(object::DataObject, data)
+    proc = Dagger.thunk_processor()
+    scope = Dagger.scope(worker=myid())
+
+    chunk = Dagger.tochunk(data, proc, scope)
+
+    object.data = chunk
+end
```

Comment:
The part with making a chunk seems to be problematic with multiple processes when `addprocs` is used (running with `julia -p` strangely seems to be fine)

Is it necessary to have data as a chunk if what the algorithm receives is a whole `DataObject`? If I understood correctly the motivation was to place the chunk on the local worker, but as is we have to transfer a whole `DataObject` as it may live somewhere else.

I tried removing explicit chunk creation and relay on Dagger to handle it, and ended up with something like:
```jl
mutable struct DataObject
    data
    size::Float64
end

function populate_data_object!(object::DataObject)
    object.data = " "^round(Int, object.size)
end
```
which seems to work without problems with `addprocs` and `-p`. I think we could aim now for minimal version and optimize transfers later

Reply:
As I understand it, the Chunk acts as a reference to some data, so moving around a DataObject with a Chunk in it is just moving around a struct that contains a reference rather than moving around the data itself. The actual data should only get transferred when you collect the Chunk in an algorithm.

I agree we can do without it for now.